CMakeLists.txt

```diff
@@ -407,8 +448,10 @@ set(Boost_ADDITIONAL_VERSIONS
 #
 # TODO for the brave: transition all mlpack's CMake to 'target-based modern
 # CMake'.  Good luck!  You'll need it.
-set(Boost_NO_BOOST_CMAKE 1)
-find_package(Boost "${BOOST_VERSION}")
+if(NOT CMAKE_CROSSCOMPILING)
+  set(Boost_NO_BOOST_CMAKE 1)
+  find_package(Boost "${BOOST_VERSION}")
```

Comment:
In fact, if you set BOOST_ROOT, you can still call `find_package()` here just fine.  In our case we no longer link with any Boost libraries directly, so we only need to find the header. :+1:

Reply:
Same as above :(

Reply:
Same as above :(